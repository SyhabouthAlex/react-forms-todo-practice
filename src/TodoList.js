import React, {useState} from 'react';
import './TodoList.css';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

const TodoList = ({todoArray}) => {
    const array = todoArray ? todoArray : [];
    const [todos, setTodos] = useState(array);
    const addTodo = (formData) => {
        setTodos(todos => [...todos, {...formData}]);
    };
    const removeTodo = (idx) => {
        setTodos(todos => todos.filter((v, i) => i !== idx));
    };

    return (
        <div className="TodoList">
            <h2 className="TodoList-title">Todos!</h2>
            <NewTodoForm addForm={addTodo}/>
            <div className="TodoList-todos">
                {todos.map(({task}, idx) => (
                    <Todo key={idx} task={task} removeTodo={() => removeTodo(idx)}/>
                ))}
            </div>
        </div>
    );
};

export default TodoList;