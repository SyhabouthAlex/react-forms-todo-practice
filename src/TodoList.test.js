import React from "react";
import {render, fireEvent} from "@testing-library/react"
import TodoList from "./TodoList"

it("renders without crashing", function() {
    render(<TodoList/>);
});

it("matches snapshot", function() {
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("shows the form", function() {
    const {getByText} = render(<TodoList/>);
    const formText = getByText("Task");
    expect(formText).toBeInTheDocument();
})

it("is able to create a new box when completing the form", function() {
    const {getByLabelText, queryByText} = render(<TodoList/>);

    const taskInput = getByLabelText("Task");
    const submitBtn = queryByText("Create a new todo!")

    fireEvent.change(taskInput, {target: {value: "banana"}});
    fireEvent.click(submitBtn);

    expect(queryByText("banana")).toBeInTheDocument();
})