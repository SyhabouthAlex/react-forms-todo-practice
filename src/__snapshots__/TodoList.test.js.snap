// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h2
      class="TodoList-title"
    >
      Todos!
    </h2>
    <form
      class="NewTodoForm"
    >
      <label
        class="NewTodoForm-input"
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        placeholder="Task"
        type="text"
        value=""
      />
      <button
        class="NewTodoForm-button"
      >
        Create a new todo!
      </button>
    </form>
    <div
      class="TodoList-todos"
    />
  </div>
</DocumentFragment>
`;
